### PR TITLE
[optimizer] Add optimizer section to INI @open sesame 04/01 17:19

### PR DESCRIPTION
**Pull request by kparichay**
Add a new optimizer section to INI
This is essential to allow custom optimizers as the
properties of optimizers needs to be iterables.
Currently, optimizer needs to specified in the model, which
along being not the best design, limits the properties to
be set for the optimizer as the model loader has to know
which properties are for optimizer and which for model.

This patch separates the optimizer as a new section and updates
the unittests.

V2:
Added backward compatibility for optimizers.
Update ini loader for model to support more properties
than the ones which are hardcoded.

See also #986

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/models/model_loader.cpp [hunk 1]
```diff
@@ -70,61 +98,6 @@ int ModelLoader::loadModelConfigIni(dictionary *ini, NeuralNetwork &model) {
   model.batch_size =
     iniparser_getint(ini, "Model:Batch_Size", model.batch_size);
 
-  /** Default to adam optimizer */
-  const char *opt_type = iniparser_getstring(ini, "Model:Optimizer", "adam");
```
Comment (zhoonit) [inline nntrainer/models/model_loader.cpp:74]: I think we should provide backward compatibility. There are number of people trying our library.

Comment (zhoonit) [reply nntrainer/models/model_loader.cpp:74]: Would this be included to tizen 6.5m1? How about we change to tizen docs as well?

Comment (kparichay) [reply nntrainer/models/model_loader.cpp:74]: will add backward compatibility.

**Comment by zhoonit** [comment]
removing release critical for now

**Comment by zhoonit** [comment]
LGTM. Could you add one test that contains old model format?

**Comment by kparichay** [comment]
> LGTM. Could you add one test that contains old model format?

In `unittest_nntrainer_models`, only some of the ini models are update to match the updated model format. Others are still using the deprecated version.

**Review by zhoonit — APPROVED** [review]

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**